src/main/java/Duke.java

```diff
@@ -1,10 +1,236 @@
+     * @return result of the command
+     */
+    public void deadline(ArrayList<Task>list,String s) throws ErrorException{
+        if (s.length() == 8) {
+            throw new ErrorException("☹ OOPS!!! The description of a todo cannot be empty.");
+        }
+        else {
+            int sIndex = s.indexOf("/");
+            int firstIndex = sIndex - 1;
+            int secondIndex = sIndex + 4;
+            String deadlineDescription = s.substring(9, firstIndex);
+            String deadlineBy = s.substring(secondIndex);
+            Deadline d = new Deadline(deadlineDescription, deadlineBy);
+            tasks.timeform(deadlineBy,d);
+        }
+    }
+    /**
+     * Executes the command and returns the result.
+     * @param s user command
+     * @return result of the command
+     */
+    public void event(ArrayList<Task>list,String s) throws ErrorException{
+        if (s.length() == 5) {
+            throw new ErrorException("☹ OOPS!!! The description of a todo cannot be empty.");
+        }
+        else {
+            int sIndex = s.indexOf("/");
+            int firstIndex = sIndex - 1;
+            int secondIndex = sIndex + 4;
+            String eventDescription = s.substring(6, firstIndex);
+            String eventAt = s.substring(secondIndex);
+            Event event = new Event(eventDescription, eventAt);
+            tasks.timeform(eventAt,event);
+
+
+
+        }
+    }
+    /**
+     * Executes the command and returns the result.
+     * @param s user command
+     * @return result of the command
+     */
+    public void other(String s) throws ErrorException{
+        throw new ErrorException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+    }
+    /**
+     * Executes the command and returns the result.
+     * @param p user command
+     * @return result of the command
+     */
+    public void delete(ArrayList<Task>list,int p){
+        Task rt = list.get(p);
+        ui.drawline();
+        System.out.println("     Noted. I've removed this task:");
+        System.out.println("     "+rt);
+        list.remove(p);
+        System.out.println("     Now you have " + list.size() + " tasks in the list.");
+        ui.drawline();
+        storage.addtask(tasks.getlist());
+    }
+    /**
+     * Executes the command and returns the result.
+     */
+    public void list(ArrayList<Task>list){
+        ui.drawline();
+        System.out.println("     Here are the tasks in your list:");
+        tasks.showlist();
+        ui.drawline();
+    }
+    /**
+     * Executes the command and returns the result.
+     */
+    public void echo(ArrayList<Task>list,Task t){
+        list.add(t);
+        ui.drawline();
+        System.out.println("     added: "+t.getName());
+        ui.drawline();
+    }
+    /**
+     * Executes the command and returns the result.
+     * @param k user command
+     * @return result of the command
+     */
+    public void done(ArrayList<Task>list,int k){
```

Comment:
`execute` might be a better name for this function

Reply:
Thank you for the advise, I will take care of that